drivers/sensor/bmm150/bmm150.c

```diff
@@ -0,0 +1,602 @@
+		return -EIO;
+	}
+
+	data->rep_z = val;
+
+	return 0;
+}
+#endif
+/* Reference Datasheet 4.3.2 */
+static s32_t bmm150_compensate_xy(struct bmm150_trim_regs *tregs,
+				    int16_t xy, uint16_t rhall, bool is_x)
+{
+	s8_t txy1, txy2;
+	s16_t val;
+	
+	if(xy == BMM150_XY_OVERFLOW_VAL) {
+		return INT32_MIN;
+	}
+
+	if (!rhall) {
+		rhall = tregs->xyz1;
+	}
+
+	if(is_x) {
+		txy1 = tregs->x1;
+		txy2 = tregs->x2;
+	} else {
+		txy1 = tregs->y1;
+		txy2 = tregs->y2;
+	}
+
+	val = ((s16_t)(((u16_t)((((s32_t)tregs->xyz1) << 14) / rhall)) -
+              ((u16_t)0x4000)));
+        val = ((s16_t)((((s32_t)xy) * ((((((((s32_t)tregs->xy2) *
+              ((((s32_t)val) * ((s32_t)val)) >> 7)) + (((s32_t)val) *
+              ((s32_t)(((s16_t)tregs->xy1) << 7)))) >> 9) +
+              ((s32_t)0x100000)) * ((s32_t)(((s16_t)txy2) +
+              ((s16_t)0xA0)))) >> 12)) >> 13)) + (((s16_t)txy1) << 3);
+
+	return (int32_t)val;
+
+}
+
+static s32_t bmm150_compensate_z(struct bmm150_trim_regs *tregs,
+				   s16_t z, u16_t rhall)
+{
+	s32_t val;
+
+	if (z == BMM150_Z_OVERFLOW_VAL) {
+		return INT32_MIN;
+	}
+
+	val = (((((s32_t)(z - tregs->z4)) << 15) - ((((s32_t)tregs->z3) *
+              ((s32_t)(((s16_t)rhall) - ((s16_t)tregs->xyz1)))) >> 2)) /
+              (tregs->z2 + ((s16_t)(((((s32_t)tregs->z1) *
+              ((((s16_t)rhall) << 1))) + (1 << 15)) >> 16))));	
+
+	return val;
+}
+
+static int bmm150_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+
+ 	struct bmm150_data *drv_data = dev->driver_data;
+	const struct bmm150_config *config = dev->config->config_info;
+	u16_t values[BMM150_AXIS_XYZR_MAX];
+	s16_t raw_x, raw_y, raw_z;
+	u16_t rhall;
+
+ 	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_MAGN_ANY);
+
+	if(i2c_burst_read(drv_data->i2c, config->i2c_slave_addr,
+				BMM150_REG_X_L, (u8_t *)values,
+				sizeof(values)) < 0) {
+		SYS_LOG_DBG("failed to read sample");
+		return -EIO;
+	}
+ 
+	raw_x = (s16_t)sys_le16_to_cpu(values[BMM150_AXIS_X]) >>
+		BMM150_SHIFT_XY_L;
+	raw_y = (s16_t)sys_le16_to_cpu(values[BMM150_AXIS_Y]) >>
+		BMM150_SHIFT_XY_L;
+	raw_z = (s16_t)sys_le16_to_cpu(values[BMM150_AXIS_Z]) >>
+		BMM150_SHIFT_Z_L;
+
+	rhall = sys_le16_to_cpu(values[BMM150_RHALL]) >>
+		BMM150_SHIFT_RHALL_L;
+
+	drv_data->sample_x = raw_x;
+	drv_data->sample_y = raw_y;
+	drv_data->sample_z = raw_z;
+	drv_data->rhall = rhall;
+
+	return 0;
+}
+
+static void bmm150_convert(struct sensor_value *val, int raw_val)
+{
+	/* val = raw_val /1600 */
+	val->val1 = raw_val / 1600;
+	val->val2 = ((s32_t)raw_val * (1000000 / 1600)) % 1000000;
+}
+
+static int bmm150_channel_get(struct device *dev, 
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct bmm150_data *drv_data = dev->driver_data;
+        
+	switch (chan) {
+	case SENSOR_CHAN_MAGN_X:
+		drv_data->sample_x = bmm150_compensate_xy(&drv_data->tregs, 
+				drv_data->sample_x,drv_data->rhall,
+                                                  true);
```

Comment:
here, as you cannot align the parameters to last opened '(' (due to 80 chars limit), you still need to align subsequent parameters to each other: so 'true' should be below drv_data->sample_...

Apply that where relevant.